devito/function.py

```diff
@@ -636,48 +649,24 @@ class TimeFunction(Function):
 
     def __init__(self, *args, **kwargs):
         if not self._cached():
-            super(TimeFunction, self).__init__(*args, **kwargs)
-            self.time_dim = kwargs.get('time_dim', None)
-            self.save = kwargs.get('save', None)
-
-            time_order = kwargs.get('time_order', 1)
-            if isinstance(time_order, int):
-                self.time_order = time_order
-                self._halo = ((time_order, 0),) + self._halo
-            elif isinstance(time_order, tuple) and len(time_order) == 3:
-                time_order, left_points, right_points = time_order
-                self.time_order = time_order
-                self._halo = ((left_points, right_points),) + self._halo
-            else:
-                raise ValueError("'space_order' must be int or 3-tuple of ints")
+            super(TimeFunction, self).__init__(**kwargs)
 
-            self._padding = (kwargs.get('time_padding', (0, 0)),) + self._padding
+            # Check we won't allocate too much memory for the system
+            available_mem = virtual_memory().available
+            if np.dtype(self.dtype).itemsize * self.size > available_mem:
```

Comment:
note for reviewers: using ``self.size`` instead of ``self.save`` should be intended as a bug fix